modules/src/core/ics04_channel/events.rs

```diff
@@ -508,41 +546,66 @@ impl EventType for OpenConfirm {
 }
 
 #[derive(Debug, Deserialize, Serialize, Clone, PartialEq, Eq)]
-pub struct CloseInit(pub Attributes);
+pub struct CloseInit {
+    pub height: Height,
+    pub port_id: PortId,
+    pub channel_id: ChannelId,
+    pub connection_id: ConnectionId,
+    pub counterparty_port_id: PortId,
+    pub counterparty_channel_id: Option<ChannelId>,
+}
 
 impl CloseInit {
+    pub fn attributes(&self) -> Attributes {
```

Comment:
imho this method should take ownership of `self` rather than perform a bunch of clones hidden to the caller.

Reply:
Alternatively, we could implement `From<CloseInit> for Attributes` for these events and remove this method.

This would make a nice parallel to the `Protobuf` instances we have all over the codebase where we have: `SomeDomainType: TryFrom<RawProtobufType>` and `RawProtobufType: From<SomeDomainType>`.

What do you think @hu55a1n1?

Reply:
I think that's a great idea. 👍 I just looked up usages of the `attributes()` method and it seems we usually don't use the event after calling this function, so it's good to consume the event.

Reply:
Okay, I'll do that.